packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -70,6 +71,12 @@ class FlutterPlugin implements Plugin<Project> {
             throw new GradleException("flutter.sdk must point to the Flutter SDK directory")
         }
 
+        String flutterExecutableName = localProperties.getProperty("flutter.executable")
+        if (flutterExecutableName == null) {
+            throw new GradleException("flutter.executable must be set to the name of the flutter executable")
+        }
+        flutterExecutable = Paths.get(flutterRoot.absolutePath, "bin", flutterExecutableName).toFile();
```

Comment:
I'd put the OS check here, so we don't need to add an extra property to local.properties. But either way works.